### ISSUE TITLE
Add option to not overwrite xorg DPI / resolution settings for HD / FullHD monitors

### ISSUE THREAD

**Issue by igpg** [linuxmint/Cinnamon#4033]
----------------------------------------------------
Important / problem background
----------------------------------------------------
Currently there is a well known Xorg problem where Xorg defaults to 96 dpi,
https://bugs.freedesktop.org/show_bug.cgi?id=41115
Which currently it's forwarded in to the dustbin by Xorg lasy developers as they are to lasy to code and support the different new resolutions. Or maybe they just decided to stick with the old display era and not support new HD / FullHD / Retina , etc displays. Kind of Donkey-Xote if you ask me.

To make a long story short, they simply don't support Full HD or HD screens and the display is all messed up (extremely small), this is because there is no code to support that. This is a well known issue also on gnome (they default to 96) and other DE / WM's, however there is the option to force Xorg to a specific resolution and DPI, which as a result works on tolreant VM / DE's which respect the values set by Xorg like LXDE (which I'm using) or XFCE which doesn't support at first site however it has an option to manually set the DPI which is equivalent to the "Text Scaling" of Cinnamon which has the following problem:

It increases only the text and not the icons, and when it's "to much for it" everything transforms bold (which is not bearable), and not all text is increased in size.

I can't use HiDPI as my screen is a FullHD so it's not HiDPI.

---------------------------------------------------
Now the problem:
---------------------------------------------------

Basically Cinnamon needs to add an option to not overwrite or ignore Xorg DPI which are set via xrandr or xorg conf files.

This are the correct settings of my screen which are working well on LXDE and Opera Browser (which does not overwrite the DPI)

 cat /etc/X11/xorg.conf.d/90-monitor.conf 
Section "Monitor"
    Identifier             "eDP1"
    DisplaySize            345 194   
    Option                 "DPI" "144 x 144"
EndSection

**Comment by igpg** [comment]
hi guys , 

any hope this will get fixed ?

**Comment by igpg** [comment]
A comment would be really appreciated

**Comment by mtwebster** [comment]
I thought I had commented.. apparently I never hit send, sorry.

You can override things using a gsettings key - see https://github.com/linuxmint/cinnamon-settings-daemon/blob/master/plugins/xsettings/README.xsettings

if you wanted to override Xft/DPI you could run dconf-editor, navigate to org.cinnamon.settings-daemon.plugins.xsettings *overrides* key, and put in there:

assuming 144 * 1024 per the document

```
'Xft/DPI': <147456>
```
(in between the brackets { } )

Not sure this is what you're after or not,

**Comment by mtwebster** [comment]
I doubt going find a full solution here.. Cinnamon shares essentially the same shortcomings as gnome-shell - and uses the same libraries.

If I read you correctly, you have an unreadably-small 1080 monitor, that you want to scale up slightly?

You're never going to get icons/images to nicely size with fonts - I imagine you'd have to used some Xrandr hacks to get what you want, and I'd be willing to wager it's broken (and not going to be fixed)

**Comment by Vahan86** [comment]
@igpg, is this still an issue for you with Cinnamon 3.4?

**Comment by mtwebster** [comment]
Xrandr can be turned off in Startup Programs now. This may serve.

**Comment by notgood** [comment]
Here are my lightdm.conf settings 
xserver-command=X -dpi 157
xft-dpi=157

(Both of these settings can be specified in other Xorg configs /etc/X11/*, same result really)

Cinnamon still completely ignores and (which is even worse), brutally overwrites both of these DPI settings:
1) Xrandr settings daemon always sets Xorg screen DPI to 96, on startup and everytime something is changed in Display applet (turning Xrandr daemon startup off as suggested obviosly breaks this applet). And 96 DPI makes every QT5 application tiny on HiDPI displays, since QT scales according to Xorg screen DPI.
Possible solution: calculate and set Xorg screen DPI according to xrandr provided display width/height in mm and pixels resolution, and only fallback to 96dpi in case of multiple (and differently sized monitors) or incorrect (zero) width/height data.

2) Fonts applet setting org.cinnamon.desktop.interface.text-scaling-factor always overwrite Xft.dpi setting, ignoring upstream supplied value. Breaks GDK/GTK & co applications, requiring user to mess with Cinnamon settings trying to guess scaling factor, when in fact correct DPI was already configured.
Possible solution: don't touch Xft.dpi unless org.cinnamon.desktop.interface.text-scaling-factor is set to non-default value, i.e. user actually changes something.

**Comment by igpg** [comment]
Sorry guys, I lost hope on this from 2015, I'v switched since to Mate which works just fine.

```
cat /etc/X11/xorg.conf.d/90-monitor.conf 
Section "Monitor"
    Identifier             "eDP1"
    DisplaySize            345 194   
    Option                 "DPI" "144 x 144"
EndSection
```
